builtins/amp-img.js

```diff
@@ -65,8 +65,21 @@ export class AmpImg extends BaseElement {
   }
 
   /** @override */
+  preconnectCallback(onLayout) {
+    // NOTE(@wassgha) the correct src url isn't determined until the element is
+    // laid out, thus assuming that hopefully all urls in the srcset/src
+    // belong to the same domain, we take the first url and pre-connect to it.
+    // If the image doesn't have an srcset, this connects to the src url.
+    this.preconnect.url(this.srcset_[0].url, onLayout);
+  }
+
+  /** @override */
   buildCallback() {
     this.isPrerenderAllowed_ = !this.element.hasAttribute('noprerender');
+
+    if (!this.srcset_) {
```

Comment:
There isn't a need to check `if (!this.srcset_)`, it should always be `null` (when this was in `layoutCallback`, situation was different since `layoutCallback` could be called again after an `unlayoutCallback`)